llvm/include/llvm/ADT/AddressRanges.h

```diff
@@ -124,8 +127,8 @@ class AddressRanges : public AddressRangesBase<AddressRange> {
     if (Range.empty())
       return Ranges.end();
 
-    auto It = llvm::upper_bound(Ranges, Range);
-    auto It2 = It;
+    auto *It = upper_bound(Ranges, Range);
+    auto *It2 = It;
```

Comment:
How do we know this is always a pointer?

Reply:
You're right I just checked and there's no guarantee they'll be pointers, I merely interpreted that linked rule as adding it regardless of that guarantee. Shall I revert this change?

Reply:
yes